shacl12-core/index.html

```diff
@@ -3000,6 +3000,66 @@ <h4>sh:pattern</h4>
 					</aside>
 				</section>
 				
+				<section id="SingleLineConstraintComponent">
+					<h4>sh:singleLine</h4>
+					<p>
+						When set to <code>true</code>, <code>sh:singleLine</code> specifies that none of the <a>value nodes</a> must contain line breaks.
+						In addition to constraint validation, this information can be exploited by user interface builders to select between (single-lined) text fields and (multi-lined) text areas.
+					</p>
+					<p>
+						<span class="component-class">Constraint Component IRI</span>: <code>sh:SingleLineConstraintComponent</code>
+					</p>
+
+					<div class="parameters">Parameters:</div>
+					<table class="term-table">
+						<tr>
+							<th>Property</th>
+							<th>Summary and Syntax Rules</th>
+						</tr>
+						<tr>
+							<td><code>sh:singleLine</code></td>
+							<td>
+								<code>true</code> to activate this constraint.
+								<span data-syntax-rule="singleLine-datatype">The values of <code>sh:singleLine</code> in a shape are literals with datatype <code>xsd:boolean</code>.</span>
+								<span data-syntax-rule="singleLine-maxCount">A shape has at most one value for <code>sh:singleLine</code>.</span>
+							</td>
+						</tr>
+					</table>
+					<div class="def def-text">
+						<div class="def-header">TEXTUAL DEFINITION</div>
+						<div class="def-text-body" data-validator="SingleLine">
+							If <code>$singleLine</code> is <code>true</code> then for each <a>value node</a> that is a literal where the lexical form matches the
+							regular expression <code>[\r\n]</code> there is a validation result.
```

Comment:
There are two Unicode "line terminators" in the ASCII subset of Unicode:

[Form Feed](https://en.wikipedia.org/wiki/Form_Feed): U+000C (sometimes `\f`)

[Vertical Tab](https://en.wikipedia.org/wiki/Vertical_Tab): U+000B (sometimes `\v`)